include/maidsafe/crux/detail/header.hpp

```diff
@@ -55,10 +55,10 @@ struct handshake {
     }
 
     void encode(detail::encoder& encoder) const {
-        encoder.put<std::uint16_t>(header::constant::type_handshake
-                                   | std::min<std::size_t>(3, retransmission_count)
-                                   | (ack ? header::constant::ack_type_cumulative
-                                          : header::constant::ack_type_none));
+        encoder.put<std::uint16_t>(
+            header::constant::type_handshake
+            | static_cast<std::uint16_t>(std::min<std::size_t>(3, retransmission_count))
```

Comment:
No, it still emits

> warning C4267: 'argument' : conversion from 'size_t' to 'const uint16_t', possible loss of data

which I think is correct.  In theory, `std::min<std::uint16_t>` would cast `retransmission_count` to a `uint16_t` which could yield in an incorrect result if `retransmission_count` is greater than the max value of a `uint16_t` wouldn't it?  I think it's safer to do the cast after we have the result and are guaranteed to be within the limits of a `uint16_t`.